src/main/java/org/folio/mr/service/flow/BatchSplitProcessor.java

```diff
@@ -44,11 +80,127 @@ public void onError(BatchSplitContext context, Exception exception) {
       .map(cause -> exception.getMessage() + ", cause: " + cause.getMessage())
       .orElse(exception.getMessage());
 
-    log.error("Batch split entity with id: {} has failed with error: {}",
+    log.error("onError:: Batch split entity with id: {} has failed with error: {}",
       splitEntity.getId(), errorMessage);
 
     splitEntity.setErrorDetails(errorMessage);
     splitEntity.setStatus(FAILED);
     batchRequestSplitRepository.save(splitEntity);
   }
+
+  private void createRequest(MediatedBatchRequest batchRequest, MediatedBatchRequestSplit splitEntity,
+                             EnvironmentType envType) {
+    if (envType == EnvironmentType.ECS) {
+      log.info("createRequest:: creating ECS request");
+      createEcsRequest(batchRequest, splitEntity);
+      return;
+    } else if (envType == EnvironmentType.SINGLE_TENANT) {
+      log.info("createRequest:: creating Single Tenant request");
+      createSingleTenantRequest(batchRequest, splitEntity);
+      return;
```

Comment:
The return statement is present in upper condition, I don't think that else if is necessary here